service/src/main/kotlin/xyz/lebkuchenfm/domain/eventstream/EventStream.kt

```diff
@@ -1,5 +1,33 @@
 package xyz.lebkuchenfm.domain.eventstream
 
-interface EventStream {
-    suspend fun sendToEveryone(event: Event)
+import io.ktor.util.collections.ConcurrentMap
+import java.util.UUID
+
+abstract class EventStream<ConsumerT : EventStream.Consumer> {
+    protected val subscriptions: MutableMap<EventStreamConsumerId, ConsumerT> = ConcurrentMap()
+
+    abstract suspend fun sendToOne(id: EventStreamConsumerId, event: Event)
+
+    suspend fun sendToEveryone(event: Event, exclude: EventStreamConsumerId? = null) {
+        for (id in subscriptions.keys) {
+            if (id == exclude) continue
+            sendToOne(id, event)
+        }
+    }
+
+    open fun subscribe(consumer: ConsumerT) {
+        subscriptions[consumer.id] = consumer
+    }
+
+    open fun unsubscribe(consumer: ConsumerT) {
+        subscriptions.remove(consumer.id)
+    }
+
+    val subscriptionCount get() = subscriptions.count()
```

Comment:
do we need that?

Reply:
Not in this branch, but it's used in my branch. I can remove it for now.